### PR TITLE
Update Apple client with changes from demoable build

### PR DESCRIPTION
**Pull request by jamilbk**
Brings in the changes from the Demoable build so I can start getting feedback from users on.

### CODE REVIEW

File: swift/apple/FirezoneNetworkExtension/PacketTunnelProvider.swift [hunk 1]
```diff
@@ -14,9 +16,13 @@ enum PacketTunnelProviderError: String, Error {
 }
 
 class PacketTunnelProvider: NEPacketTunnelProvider {
-  static let logger = Logger(subsystem: "dev.firezone.firezone", category: "packet-tunnel")
+  let logger = Logger(subsystem: "dev.firezone.firezone", category: "packet-tunnel")
 
-  private lazy var adapter = Adapter(with: self)
+  // Till connlib is updated to expose the desired FFI, we'll use a mock
+  private lazy var adapter = ConnlibMock.Adapter(with: self)
```
Comment (francesca64) [inline swift/apple/FirezoneNetworkExtension/PacketTunnelProvider.swift]: Couldn't this be made configurable or something? It's important to be able to test connlib with the Apple client; if a specific pending FFI change is blocking this I can address that ASAP.

**Review by francesca64 — CHANGES_REQUESTED** [review]

File: swift/apple/FirezoneNetworkExtension/PacketTunnelProvider.swift (hunk 1 above)
Comment (jamilbk) [reply swift/apple/FirezoneNetworkExtension/PacketTunnelProvider.swift]: That's a good question -- probably better suited for @roop. This was just copy-pasta'd from the `demoable_macos_app` branch in the firezone-apple repo.

I don't think this PR needs merging urgently; I'll mark it as draft and defer to @roop. This was mainly to get the updated UI into TestFlight, but that process is decoupled from our CI pipeline at the moment.

Comment (jamilbk) [reply swift/apple/FirezoneNetworkExtension/PacketTunnelProvider.swift]: I've reverted everything except the UI changes and it still builds fine, so I'll leave it like this and see if @roop wants to make changes.

Comment (roop) [reply swift/apple/FirezoneNetworkExtension/PacketTunnelProvider.swift]: I've added commits to make the status bar menu (macOS) update when connlib updates its resources .

**Comment by roop** [comment]
Nice work, @jamilbk 🙌 .

I added commits to show the updates in the macOS status menu. We still don't derive the network settings from what connlib gives us and apply it. That shall come in a separate PR.

**Review by roop — APPROVED** [review]

**Review by francesca64 — APPROVED** [review]

**Review by roop — APPROVED** [review]

**Review by conectado — APPROVED** [review]

**Pull request merged**